site/en/blog/timer-throttling/index.md

```diff
@@ -0,0 +1,238 @@
+---
+layout: 'layouts/blog-post.njk'
+title: Timer throttling
+authors:
+  - jakearchibald
+description: >
+  Chrome 88 will heavily throttle chained JavaScript timers for hidden pages.
+date: 2021-01-11
+hero: image/CZmpGM8Eo1dFe0KNhEO9SGO8Ok23/5diIn5KBi44iwy3o6GRi.jpg
+alt: A collection of broken clocks
+---
+
+Chrome 88 (January 2020) will heavily throttle chained JavaScript timers for
+hidden pages in particular conditions. This will reduce CPU usage, which will
+also reduce battery usage. There are some edge cases where this will change
+behavior, but timers are often used where a different API would be more
+efficient, and more reliable.
+
+Ok, that was pretty jargon heavy, and a bit ambiguous. Let's dig in…
+
+## Terminology
+
+### Hidden pages
+
+Generally, 'hidden' means a different tab is active, or the window has been
+minimized, but browsers may consider a page hidden whenever its content is
+totally not-visible. Some browsers go further than others here, but you can
+always use the [page visibility
+API](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API) to
+track when the browser thinks visibility has changed.
+
+### JavaScript timers
+
+By "JavaScript timers" I mean `setTimeout` and `setInterval`, which allow you to
+schedule a callback sometime in the future. Timers are useful, and they aren't
+going away, but sometimes they're used to poll state when an event would be more
+efficient, and more accurate.
+
+### Chained timers
+
+If you call `setTimeout` in the same task as a `setTimeout` callback, the second
+invocation is 'chained'. With `setInterval`, each iteration is part of the
+chain. This might be easier to understand with code:
+
+```js
+let chainCount = 0;
+
+setInterval(() => {
+  chainCount++;
+  console.log(`This is number ${chainCount} in the chain`);
+}, 500);
+```
+
+And:
+
+```js
+let chainCount = 0;
+
+function setTimeoutChain() {
+  setTimeout(() => {
+    chainCount++;
+    console.log(`This is number ${chainCount} in the chain`);
+    setTimeoutChain();
+  }, 500);
+}
+```
+
+Ok, I think that's all the terms you need to know.
+
+## How the throttling works
+
+The throttling happens in stages:
+
+### Minimal throttling
+
+This happens to timers that are scheduled when _any_ of the following is true:
```

Comment:
`scheduled` seems to be an important concept that you might want to add to your terminology section?

Reply:
It's already mentioned in the timers section